openfe/setup/molecule.py

```diff
@@ -20,14 +28,37 @@ class Molecule:
         will be used in the hash.
     """
     def __init__(self, rdkit: RDKitMol, name: str = ""):
-        self._rdkit = rdkit
+        # internally we store as RDKit, for now...
+        self._rdkit = Chem.Mol(rdkit)
         self._hash = hashmol(self._rdkit, name=name)
 
-    # property for immutability; also may allow in-class type conversion
-    @property
-    def rdkit(self) -> RDKitMol:
-        """RDKit representation of this molecule"""
-        return self._rdkit
+    def to_rdkit(self) -> RDKitMol:
+        """Return an RDKit copied representation of this molecule"""
+        return Chem.Mol(self._rdkit)
+
+    @classmethod
+    def from_rdkit(cls, rdkit: RDKitMol, name: str = ""):
+        """Create a Molecule copying the input from an rdkit Mol"""
+        return cls(rdkit=Chem.Mol(rdkit), name=name)
+
+    def to_openeye(self) -> OEMol:
+        """OEChem representation of this molecule"""
+        return self.to_openff().to_openeye()
+
+    @classmethod
+    def from_openeye(cls, oemol: OEMol, name: str = ""):
+        raise NotImplementedError
+
+    def to_openff(self) -> OFFMolecule:
+        """OpenFF Toolkit representation of this molecule"""
+        m = OFFMolecule(self._rdkit, allow_undefined_stereo=True)
+        m.name = self.name
+
+        return m
+
+    @classmethod
+    def from_openff(cls, openff: OFFMolecule, name: str = ""):
+        raise NotImplementedError
```

Comment:
@dwhswenson so what I've done here is everything is a copy.  So on creation, we copy the rdkit Mol, and all to_X are copies.  The Molecule is then therefore read-only (and documented as such).  Handling manipulation of a Molecule is out of scope for us (i.e. if you want to change a Molecule, recreate it).

Reply:
100% agree with the approach. Reviewing details now.